app/src/main/java/ch/epfl/skysync/models/calendar/CalendarModel.kt

```diff
@@ -109,4 +110,22 @@ abstract class CalendarModel<T : CalendarViewable> {
   fun getByDate(date: LocalDate, timeSlot: TimeSlot): T? {
     return cells.firstOrNull { it.date == date && it.timeSlot == timeSlot }
   }
+
+  /** Return a set of the cells, sorted by date and time slot */
+  private fun getSortedSetOfCells(): SortedSet<T> {
```

Comment:
we might consider at some point to directly implement and maintain the underlying list in `CalendarModel` as a sorted structure